spec/decorators/telegram_message_spec.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe TelegramMessage do
+  let(:telegram_message) { TelegramMessage.new message }
+  describe '#text' do
+    subject { telegram_message.text }
+
+    describe 'given a message with a `text` attribute' do
+      let(:message) { { text: 'Ich bin eine normale Nachricht' } }
+      it { should eq('Ich bin eine normale Nachricht') }
+    end
+
+    describe 'given a photo with a `caption`' do
+      let(:message) { { caption: 'Das hier ist die Überschrift eine Fotos' } }
+      it { should eq('Das hier ist die Überschrift eine Fotos') }
+    end
+  end
+
+  describe '#photos' do
+    subject { telegram_message.photos }
+    describe 'given a message without photos' do
+      let(:message) { { text: 'Ich bin eine normale Nachricht' } }
+      it { should eq([]) }
+    end
+
+    describe 'given a message with one photo', vcr: { cassette_name: :photo_with_image } do
+      let(:message) { message_with_photo }
+
+      it { should_not be_empty }
+      it { should all(be_a(Photo)) }
+
+      it 'chooses the largest image' do
+        photo = subject.first
+        expect(photo.image.blob.byte_size).to eq(90_449)
+      end
+    end
+  end
+
+  describe '#reply', vcr: { cassette_name: :photo_with_image } do
+    let(:request) { create(:request) }
+    let(:message) { message_with_photo }
+    subject { telegram_message.reply }
+    it { should be_a(Reply) }
+    describe 'assigning a request and calling #save! on the reply' do
+      it {
+        expect do
+          subject.request = request
+          subject.save!
+        end.to(change { Reply.count }.from(0).to(1))
+      }
+
+      describe 'given the user sends a series of images as album', vcr: { cassette_name: :photo_album } do
+        let(:message_with_media_group_id) { message_with_photo.merge(media_group_id: '42') }
+        let(:save_reply_and_photo) do
+          lambda {
+            tm = TelegramMessage.new message_with_media_group_id
+            reply = tm.reply
+            reply.request = request
+            reply.save
+            reply.photos << tm.photos
+          }
+        end
+
+        it { expect { 3.times { save_reply_and_photo.call } }.to(change { Reply.count }.from(0).to(1)) }
+        it { expect { 3.times { save_reply_and_photo.call } }.to(change { Photo.count }.from(0).to(3)) }
```

Comment:
Hmmm, intuitively, I’d have expected it not to save a photo a second time if it already exists.

Edit: I guess we’d have to save the `file_id` for every photo to ensure that, so I’m fine with that.